src/ruby.rs

```diff
@@ -51,100 +51,890 @@ extern {
+    ///     [documentation](https://ruby-doc.org/core-2.5.1/Math/DomainError.html)
     pub static rb_eMathDomainError: VALUE;
 
+
+    /// Convert a C string to an [`ID`](struct.id.html)
+    ///
+    /// * `cstr` - nul-terminated C string, treated as ASCII encoded
+    ///
+    /// # Defined In
+    ///
+    /// * **2.5:**
+    ///     [ruby.h](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1728) |
+    ///     [symbol.c](https://github.com/ruby/ruby/blob/v2_5_1/symbol.c#L610-L614) |
+    ///     [documentation](https://ruby-doc.org/core-2.5.1/doc/extension_rdoc.html#label-ID+or+Symbol)
     pub fn rb_intern(cstr: *const c_char) -> ID;
-    pub fn rb_intern2(ptr: *const c_char, len: c_long) -> ID;
+
+    /// Convert a C string of the given length to an [`ID`](struct.id.html)
+    ///
+    /// * `cstr` - C string, treated as ASCII encoded
+    /// * `len` - number of C chars
+    ///
+    /// # Miscellaneous
+    ///
+    /// [`rb_intern_const`](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1757-L1760)
+    /// is more or less an alias of this.
+    ///
+    /// # Defined In
+    ///
+    /// * **2.5:**
+    ///     [ruby.h](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1729)
+    ///     [symbol.c](https://github.com/ruby/ruby/blob/v2_5_1/symbol.c#L603-L607)
+    pub fn rb_intern2(cstr: *const c_char, len: c_long) -> ID;
+
+    /// Convert a string to an [`ID`](struct.id.html)
+    ///
+    /// * `string`: an instance of [`rb_cString`](static.rb_cString.html)
+    ///
+    /// # Safety
+    ///
+    /// * Behavior is undefined if you pass a `VALUE` that is not a string.
+    ///
+    /// # Defined In
+    ///
+    /// * **2.3:** [ruby.h](https://github.com/ruby/ruby/blob/v2_3_7/include/ruby/ruby.h#L1708)
+    /// * **2.4:** [ruby.h](https://github.com/ruby/ruby/blob/v2_4_4/include/ruby/ruby.h#L1728)
+    /// * **2.5:**
+    ///     [ruby.h](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1730)
+    ///     [symbol.c](https://github.com/ruby/ruby/blob/v2_5_1/symbol.c#L616-L626) |
+    ///     [documentation](https://ruby-doc.org/core-2.5.1/doc/extension_rdoc.html#label-ID+or+Symbol)
+    /// * **2.6:** [ruby.h](https://github.com/ruby/ruby/blob/v2_6_0_preview2/include/ruby/ruby.h#L1747)
     pub fn rb_intern_str(string: VALUE) -> ID;
 
+    /// Convert a [`rb_cSymbol`](static.rb_cSymbol.html)
+    /// to an [`ID`](struct.id.html)
+    ///
+    /// * `symbol`: an instance of [`rb_cSymbol`](static.rb_cSymbol.html)
+    ///
+    /// # Safety
+    ///
+    /// ## Exceptions
+    ///
+    /// * [`rb_eTypeError`](static.rb_eTypeError.html)
+    ///     * if `symbol` is not a `rb_cSymbol`
+    ///
+    /// # Miscellaneous
+    ///
+    /// [`RB_SYM2ID`](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L381) and
+    /// [`SYM2ID`](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L386) are currently
+    /// aliases for this.
```

Comment:
FWIW usually `RB_SYM2ID` > `SYM2ID` > `rb_sym2id` (the macros usually do more optimizations in C, but those are not always possible through FFI)